src/lib/dbus.ts

```diff
@@ -1,37 +1,50 @@
-import * as Bluebird from 'bluebird';
 import * as dbus from 'dbus';
+import { promisify } from 'util';
 import { TypedError } from 'typed-error';
+import _ = require('lodash');
 
 import log from './supervisor-console';
 
 export class DbusError extends TypedError {}
 
-const bus = dbus.getBus('system');
-const getInterfaceAsync = Bluebird.promisify(bus.getInterface, {
-	context: bus,
-});
+const getBus = _.memoize((dbusLib: typeof dbus) => dbusLib.getBus('system'));
+const getInterfaceAsync = _.memoize((bus: ReturnType<typeof getBus>) =>
+	promisify(bus.getInterface.bind(bus)),
+);
+
+function getInterface(
+	serviceName: string,
+	objectPath: string,
+	interfaceName: string,
+) {
+	return getInterfaceAsync(getBus(dbus))(
+		serviceName,
+		objectPath,
+		interfaceName,
+	);
+}
 
 async function getSystemdInterface() {
 	try {
-		return await getInterfaceAsync(
+		return await getInterface(
 			'org.freedesktop.systemd1',
 			'/org/freedesktop/systemd1',
 			'org.freedesktop.systemd1.Manager',
 		);
 	} catch (e) {
-		throw new DbusError(e);
```

Comment:
They are of type `DBusError` (note the capital B in DBus), which requires a name and a message on instantiation:
```
export class DBusError extends Error {
    readonly dbusName: string;
    constructor(name: string, message: string);
}
```

I believe the `export class DbusError extends TypedError {}` line in lib/dbus was to avoid this.